### DISCUSSION TITLE
Guidance for Staying Updated with Bitplatform, While Implementing Custom Changes.

### DISCUSSION THREAD

**Discussion by zandiarash** [bitfoundation/bitplatform#9360] [Q&A] [answered]
I am currently using Bitplatform Blazor templates for my projects and have been highly impressed with the flexibility and functionality they provide. However, as the templates are frequently updated on GitHub, I find it challenging to stay updated with the latest changes while implementing custom modifications to the core project to meet my specific needs.

To ensure I can maintain both the integrity of my custom changes and benefit from the latest updates, I would greatly appreciate your guidance on the best practices for:
1. Tracking and integrating upstream updates from the repository.
2. Managing potential conflicts between my customizations and the updates.
3. Testing and validating after applying updates to ensure compatibility.

Additionally, if there are specific tools, workflows, or documentation that you recommend for this process, I would be grateful for your recommendations.

Thank you for providing such a valuable resource for developers, and I look forward to your advice on how to manage updates effectively while maintaining customized changes.

**Comment by yasmoradi** [comment, marked as answer]
The biometric authentication, alongside with touch features like pull to refresh and swipe, and support for aspire and health checks are going to be added to Boilerplate, and due nature of being a template, there's nothing more you can do about that. We've tried to keep things in separated Nuget packages, like butil, bswup and blazorui itself, but it's not possible for entire code base 🤕
The good news is our focus is changing to build a low code tool on top of Boilerplate, so being updated will be a lot easier because of lower number of upcoming changes on Boilerplate itself.

**Reply by zandiarash** [reply]
> The biometric authentication, alongside with touch features like pull to refresh and swipe, and support for aspire and health checks are going to be added to Boilerplate, and due nature of being a template, there's nothing more you can do about that. We've tried to keep things in separated Nuget packages, like butil, bswup and blazorui itself, but it's not possible for entire code base 🤕 The good news is our focus is changing to build a low code tool on top of Boilerplate, so being updated will be a lot easier because of lower number of upcoming changes on Boilerplate itself.

Wow, you always surprise us with new amazing features🎉🎉😍❤️
One more question, Does projects other than Web updates with bswup automatically as web does ? or they must update only by markets (Play Store or etc.) ?

**Reply by yasmoradi** [reply]
Windows app also gets updates automatically 💯
For the android in app updates, there's an issue on maui side, we'll come back to android, iOS and Mac updates later

**Comment by zandiarash** [comment]
Where can I find the roadmap ?
and I promise this is the last question, Can the project be published for running on Linux ?

**Reply by yasmoradi** [reply]
You can ask as many as questions you've (":
What do you mean? for the backend? or you mean Linux desktop app for the consumers?
The roadmap is being handled in Azure DevOps, so I'm afraid, we can't share it to the public

**Comment by zandiarash** [comment]
Thanks for your kindness.
I mean publishing app for linux.
What if you make a simple roadmap globally just for announcements purpose like this : 
✅Adding biometric authentication
✅Touch features like pull to refresh and swipe
✅Supporting for aspire and health checks

**Comment by yasmoradi** [comment]
Please open issue for your requirements, for example if you really need Linux app support. 

I'm not saying yours is a bad idea, but we're not gong to do it right now for several reasons

**Reply by zandiarash** [reply]
> Please open issue for your requirements, for example if you really need Linux app support.
> 
> I'm not saying yours is a bad idea, but we're not gong to do it right now for several reasons

Thank you for the great community, your patience and kindness❤️
